proposals/0107-unsaferawpointer.md

```diff
@@ -1477,34 +1477,36 @@ struct UnsafeMutableRawPointer : Strideable, Hashable, _Pointer {
   func bindMemory<T>(to: T.Type, capacity: Int) -> UnsafeMutablePointer<T>
   func assumingMemoryBound<T>(to: T.Type) -> UnsafeMutablePointer<T>
 
-  func initialize<T>(_: T.Type, with: T, count: Int = 1)
+  func initialize<T>(_: T.Type, to: T, count: Int = 1)
     -> UnsafeMutablePointer<T>
-  func initialize<T>(toContiguous: T.Type, atIndex: Int, with: T)
+  func initialize<T>(contiguous: T.Type, at: Int, to: T)
```

Comment:
I would like to distinguish initialization via raw pointer from initialization via typed pointer primarily because raw pointer initialization binds memory to a type.

Option 1:
--------

```swift
// n copies of a single value
rawPtr.initializeMemory(as: Int.self, count: n, to: x)
// read n copies out of memory at `ptrToInt`
rawPtr.initializeMemory(as: Int.self, from: ptrToInt, count: n)
```

Option 2:
--------

```swift
// n copies of a single value
rawPtr.initializeAndBind(as: Int.self, count: n, to: x)
// read n copies out of memory at `ptrToInt`
rawPtr.initializeAndBind(as: Int.self, from: ptrToInt, count: n)
```